batavia/core/exceptions.js

```diff
@@ -22,16 +22,61 @@ BaseException.prototype.toString = function() {
 }
 
 BaseException.prototype.__str__ = function() {
-    if (this.msg) {
-        return this.msg
+    if (this.args !== undefined) {
```

Comment:
This should check for length 0. That's why StopIteration.__str__() is different.